Mage/src/main/java/mage/target/targetadjustment/TargetsCountAdjuster.java

```diff
@@ -0,0 +1,48 @@
+package mage.target.targetadjustment;
+
+import mage.abilities.Ability;
+import mage.abilities.dynamicvalue.DynamicValue;
+import mage.filter.Filter;
+import mage.game.Game;
+import mage.target.Target;
+
+/**
+ * @author TheElk801, notgreat
+ */
+public class TargetsCountAdjuster implements TargetAdjuster {
+    private Target blueprintTarget = null;
+    private final DynamicValue dynamicValue;
+
+    /**
+     * Modifies the target to be X targets, where X is the dynamic value.
+     * If the ability's target has min targets of 0, it's treated as "up to X targets".
+     * Otherwise, it's exactly "X targets".
+     *
+     * @param value The number of targets
+     */
+    public TargetsCountAdjuster(DynamicValue value) {
+        this.dynamicValue = value;
+    }
+
+    @Override
+    public void adjustTargets(Ability ability, Game game) {
+        if (blueprintTarget == null) {
+            blueprintTarget = ability.getTargets().get(0).copy();
+            blueprintTarget.clearChosen();
+        }
+        Target newTarget = blueprintTarget.copy();
+        int count = dynamicValue.calculate(game, ability, null);
```

Comment:
I think that null should probably be replaced by `ability.getEffects().get(0)`.
Working on reworking The Wise Mothman, and I extract a value from `effect.getValue(VALUE_KEY)`, set during the trigger.

Reply:
7c3bbed8f for the commit adding that `TargetAdjuster` to the Mothman. Could also be a `TargetsCountAdjuster` once the `calculate`'s `source` argument is a real one.

Reply:
Good call on the calculate effect source.